### PR TITLE
Retry ctype checks

### PR DESCRIPTION
**Pull request by paulidale**
A different approach to #4219 

- [x] tests are added or updated

### CODE REVIEW

**Comment by paulidale** [comment]
@levitte ?

**Comment by levitte** [comment]
Does `isascii` give you the answer you expect on an EBCDIC system?  Has this been tested on an EBCDIC system?

Also, I wonder why you insist on trying negative values.

**Comment by paulidale** [comment]
@t-j-h has access to an EBCDIC system and I've asked for a test run to check isascii and isdigit.

Negative values are for signed chars being passed (which is common).

My old code was definitely broken.  I'm happy for this to not go in if it isn't providing any benefits.

**Comment by paulidale** [comment]
Based on the data from zOS, I've restored the negative testing: isascii() doesn't work for negatives.  It works for values up to 255.

No other changes, but a re-review is required.

**Comment by levitte** [comment]
... But other functions work with negative values?

**Comment by paulidale** [comment]
I tested isascii and isdigit on zOS.  Neither worked for negative values.
It seems reasonable to extrapolate this to the other isXXX functions.

**Comment by paulidale** [comment]
I've got complete results now, none of the ctype functions work properly for negative values on zOS.

**Comment by levitte** [comment]
What about toupper / tolower?  I noticed you left the negative values in `case_change` intact...

**Comment by paulidale** [comment]
Only EOF which toupper and tolower are defined for.

File: test/ctype_internal_test.c [hunk 1]
```diff
@@ -53,7 +58,10 @@ static struct {
     { '%', '%' },
     { '~', '~' },
     {   0,   0 },
-    { EOF, EOF }
+    { EOF, EOF },
+    { 333, 333 },
+    { -333, -333 },
+    { -128, -128 }
```
Comment (levitte) [inline test/ctype_internal_test.c]: From what you're saying, these two lines should be removed

Comment (paulidale) [inline test/ctype_internal_test.c]: Yeah, sorry I looked at the wrong branch :(

Those three should go.  The positive is too large for an unsigned char.
Commit amended.

**Comment by paulidale** [comment]
Merged.  Thanks.

**Pull request closed without merging**